Stop JWT provider from being written in non default namespace (#18325)
hashicorp/consul: .changelog/18325.txt

Reviewer: I think we are supposed to prefix ent-only changelogs with an underscore?

Author: we only do that if the PR is in the enterprise repo iirc to avoid name collision with oss

Reviewer: Right. Also the underscore controls whether the PR link is included in the CHANGELOG.md

Reviewer: This is fine

```release-note:bug

> mesh: Stop jwt providers config entry from being created in non-default namespaces.

Reviewer: nit: word-smithing/formatting

```suggestion
mesh: **(Enterprise Only)** Require that `jwt-provider` config entries are created in the `default` namespace.
```
